app/Scripts/Player.gd

```diff
@@ -76,40 +75,35 @@ func Dash():
 # Function to refill stamina
 func StaminaRefill():
 	
-	stats.Stamina = move_toward(stats.Stamina, stats.MaxStamina, stats.staminaRegen) # regen stamina
+	PlayerStats.Stamina = move_toward(PlayerStats.Stamina, PlayerStats.MaxStamina, PlayerStats.staminaRegen) # regen stamina
 	
-	if stats.Stamina != stats.MaxStamina: # if stamina is not full, start recovering stamina again
+	if PlayerStats.Stamina != PlayerStats.MaxStamina: # if stamina is not full, start recovering stamina again
 		timer.start() 
 	
 #_____________________#
 # Hurt player if enemy enters
 func _on_HurtBox_area_entered(area):
-	stats.Health -= 10 
+	PlayerStats.Health -= 10 
 
 	
 	 
 #_____________________#
 # Signal from stats,  if the health stats ever changes this function starts
-func _on_PlayerStats_healthChange():
-	
-	health = stats.Health
+func _on_PlayerStats_healthChange(value):
 	
-	if health <= 0:
-		health = 0
-		emit_signal("health_update", health)
-		queue_free()
-	else:
-		emit_signal("health_update", health)
+		emit_signal("health_update", PlayerStats.Health)
+		
 #_____________________#
 # Signal from stats,  if the stamina stats ever changes this function starts
-func _on_PlayerStats_staminaChange():
-	Stamina = stats.Stamina
-	emit_signal("ChangedStamina", Stamina)
-
+func _on_PlayerStats_staminaChange(value):
 	
+	emit_signal("ChangedStamina", PlayerStats.Stamina)
+
+	print("stamina")
```

Comment:
just curious, here it looks like we're listening to this signal just to emit another signal. Is this necessary or could we just be more direct?